1:done

falseidol opened the pull request.

falseidol committed:
1. model user, UserController done.

falseidol committed:
1. model user, UserController done.`
2. model film, FilmController done.
3.

falseidol committed:
1. model user, UserController done.`
2. model film, FilmController done.
3.

falseidol committed:
done

falseidol committed:
done2

VladimirTulsky commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/FilmDbStorage.java:
```diff
@@ -196,6 +196,48 @@ public List<Integer> getUsersFilms(List<Integer> userIds) {
+        query = "%" + query + "%";
+        if (by != null) {
+            splitter = by.split(",");
+            if (splitter.length == 2) {
```
предлагаю немного оптимизировать метод. массив splitter используется только внутри if, можно и объявить и заполнить его в одном месте String[] splitter = by.split(",");

falseidol replied:
Вообще да, забыл убрать инициализацию сплиттера(старая версия метода)

VladimirTulsky commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/FilmDbStorage.java:
```diff
@@ -196,6 +196,48 @@ public List<Integer> getUsersFilms(List<Integer> userIds) {
+                        "left join FILM_DIRECTOR fd on f.FILM_ID = fd.FILM_ID " +
+                        "left join DIRECTORS d on d.DIRECTOR_ID = fd.DIRECTOR_ID " +
+                        "where f.name ilike ? or d.name ilike ? ORDER BY fr.rating IS NULL, fr.rating DESC";
+                films = jdbcTemplate.query(sqlIfNotNull, FilmDbStorage::makeFilm, query, query);
```
здесь можно сразу делать return, как и в остальных трех блоках

falseidol replied:
Сделано

VladimirTulsky commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/FilmDbStorage.java:
```diff
@@ -196,6 +196,48 @@ public List<Integer> getUsersFilms(List<Integer> userIds) {
+                }
+            }
+        }
+        return films;
```
а тут в случае, если ни один блок не сработал возвращать Collections.emptyList(), чтобы у нас в сервис не пошло значение null

falseidol replied:
Разумное предложение

max-vassiliev commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/FilmDbStorage.java:
```diff
@@ -196,6 +196,48 @@ public List<Integer> getUsersFilms(List<Integer> userIds) {
+        List<Film> films = null;
+        String[] splitter;
+        query = "%" + query + "%";
+        if (by != null) {
```
Небольшое предложение, не настаиваю.

Мне кажется, здесь очень большой кусок кода обернут в конструкцию `if (by != null)`. Если это условие не выполняется (то есть, если `by` равен `null`), то дальше нужно долго листать, чтобы понять, что произойдет. А произойдет только то, что метод вернет `null`.

Если учесть предложение Володи чуть ниже — возвращать не `null`, а пустой список, то предлагаю в начале метода поставить условие: если `by` равен `null`, верни пустой список. После этого `if (by != null)` можно убрать. Но это, опять же, на усмотрение. Мне так удобнее читать.

А так, в целом, круто сделано всё!)

falseidol replied:
В самом деле хорошее предложение для удобства чтения, да и вообщем для будущей работы

max-vassiliev reviewed (commented):
Отлично! От меня небольшое предложение — на твое усмотрение

falseidol committed:
1: codestyle if by == null Collections.emptyList()
2: exception if condition didn't worked, return Collections.emptyList(), method searchUsingKeyWord
